### PR TITLE
The initial pull request.

### PR DESCRIPTION
**Pull request by bruskiza**
@hodgestar not ready for review

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -3,29 +3,183 @@ go.app = function() {
...
-        self.states.add('states:start', function(name) {
-            return new ChoiceState(name, {
-                question: 'Hi there! What do you want to do?',
+        self.states.add('state_welcome', function(name) {
+            var question = $("Welcome! To report a malaria case, please enter your facility code. For example, 543456");
+            var error = $("The facility code is invalid. Please enter again.");
+            return new FreeText(name, {
+                  question: question,
+                  check: function(content) {
+                      if (go.utils.is_valid_name(content) || true ) {
```
Comment (hodgestar) [inline src/app.js]: Why the `|| true` here? Can't `is_valid_name` just return `true` or `false`?

Comment (hodgestar) [inline src/app.js]: If you invert the `if` you can just do `if (!go.utils.is_valid_name(content)) { return error; }` ?

File: src/app.js [hunk 2]
```diff
@@ -3,29 +3,183 @@ go.app = function() {
...
+        //         },
+        //         next: 'state_last_period_month'
+        //     });
+        // });
+
+
+
+
+        self.states.add('state_validate_facility_code', function(name) {
+          var question = $("Please enter the patient's cellphone number, eg: 01234567890");
```
Comment (hodgestar) [inline src/app.js]: Inconsistent indentation. :)

File: src/app.js (hunk 1 above)
Comment (bruskiza) [inline src/app.js]: yeah... it's very much WIP.... I did that to make the test pass for now.

not ready for review by any stretch :1234:

**Comment by bruskiza** [comment]
@hodgestar @smn ready for review... even if the tests are not complete.

File: go-app.js [hunk 3]
```diff
@@ -5,41 +5,446 @@
...
+    },
+
+    is_valid_msisdn: function(input) {
+        // check that it is a number, starts with 0, and has at least 10 digits
+        return go.utils.check_valid_number(input) && input[0] === '0' && input.length >= 10;
+    },
+
+    check_valid_alpha: function(input) {
+        var alpha_only = new RegExp('^[A-Za-z]+$');
+        return input !== '' && alpha_only.test(input);
```
Comment (smn) [inline go-app.js:50]: 1__node__node_

Javascript makes these things exceptionally hard.

Comment (bruskiza) [inline go-app.js:50]: @smn: this is not even my file... i copied it from WCL :\

**Comment by bruskiza** [comment]
Merged, begrudgingly.

**Pull request merged**